hibernate-core/src/main/java/org/hibernate/boot/model/naming/ImplicitNamingStrategyComponentPathImpl.java

```diff
@@ -30,14 +31,17 @@ protected String transformAttributePath(AttributePath attributePath) {
 	}
 
 	public static void process(AttributePath attributePath, StringBuilder sb) {
-		if ( attributePath.getParent() != null ) {
-			process( attributePath.getParent(), sb );
-			if ( StringHelper.isNotEmpty( attributePath.getParent().getProperty() ) ) {
-				sb.append( '_' );
-			}
-		}
-
 		String property = attributePath.getProperty();
+		final AttributePath parent = attributePath.getParent();
+		if ( parent != null && !parent.getProperty().equals( "" ) ) {
+			process( parent, sb );
+			sb.append( '_' );
+		}
+		else if ( PropertyPath.IDENTIFIER_MAPPER_PROPERTY.equals( property ) ) {
+			// skip it, do not pass go
+			sb.append( "id" );
+			return;
+		}
 		property = property.replace( "<", "" );
 		property = property.replace( ">", "" );
```

Comment:
There must be a more efficient way to replace the above two statements with one without much performance cost, though I am not able to find one in our existing classes (StringUtils from apache commons seems forbidden).

Reply:
Unless I am mistaken, the '<' and '>' can only occur as the first and last in the prop-name.  Direct access to the underlying char array is usually quite fast - 

```
if ( property.charAt( 0 ) == '<' && property.charAt( property.length() -1 ) == '>' ) {
    property = property.substring( ... );
}
```